src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -209,6 +224,17 @@ internal static void ApplyValueToEnumerable<TProperty>(ref TProperty value, Json
                     ((IList<TProperty>)frame.JsonPropertyInfo.GetValueAsObject(frame.ReturnValue, options)).Add(value);
                 }
             }
+            else if (frame.IsDictionary())
+            {
+                // todo: use TryAdd and throw JsonReaderException
+                ((IDictionary<string, TProperty>)frame.ReturnValue).Add(frame.KeyName, value);
```

Comment:
If throw on duplicate keys is your selected deserialization behaviour (as there is also the possibilitity just to use the last one), you should add a test case for it.

Reply:
I will make sure https://github.com/dotnet/corefx/issues/36025 throws the proper `JsonReaderException`.